pgjdbc/src/test/java/org/postgresql/test/jdbc2/BatchedInsertReWriteEnabledTest.java

```diff
@@ -153,16 +255,39 @@ public void testBatchWithReWrittenBatchStatementWithSemiColon()
       pstmt.setString(2, "c");
       pstmt.setInt(3, 6);
       pstmt.addBatch();
-      assertTrue(
-          "Expected outcome not returned by batch execution.",
-          Arrays.equals(new int[] { Statement.SUCCESS_NO_INFO,
-              Statement.SUCCESS_NO_INFO, Statement.SUCCESS_NO_INFO },
-              pstmt.executeBatch()));
+      Assert.assertEquals(
+          "Three batched statements should be rewritten to single multi-values one, thus expecting SUCCESS_NO_INFO",
```

Comment:
Actually, this original limitation was because I did not know how code like this would behave:
INSERT INTO X VALUES (?, random(), ?)
... but it seems that a batch of these does not evaluate the function only once. So there is no problem in nesting parentheses.